### PR TITLE
[WIP] KubeVirt images reconciliation

### PR DESCRIPTION
**Pull request by mfranczy**
**What this PR does / why we need it**:
This introduces KubeVirt images reconciliation that allows users to cache standard and create custom images.

**Which issue(s) this PR fixes**:

Fixes #

**What type of PR is this?**
/kind feature

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
Add support of standard cache and custom images creation.
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

File: pkg/resources/machinecontroller/deployment.go [hunk 1]
```diff
@@ -53,7 +53,7 @@ var (
 const (
 	Name = "machine-controller"
 	// TODO: update to tagged machine-controller version.
-	Tag = "0a94443b41516398b5bd6a129589dea599f3e7b8"
+	Tag = "e7e4ba8eb535086374108ed235c2e33bab7f4292"
```
Comment (mfranczy) [inline pkg/resources/machinecontroller/deployment.go:56]: @hdurand0710 do you want to keep that commit hash?

Comment (hdurand0710) [reply pkg/resources/machinecontroller/deployment.go:56]: nope, that would need to be the one after you merge https://github.com/kubermatic/machine-controller/pull/1529

File: pkg/provider/cloud/kubevirt/image/image.go [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// reconcile is a generic mechanism to reconcile images.
+// 1- Ensures that all DataVolumes in kdv are reconciled
```
Comment (hdurand0710) [inline pkg/provider/cloud/kubevirt/image/image.go:152]: why 1- ? 
Is a "2-" missing ?

File: pkg/webhook/seed/validation_ce.go [hunk 3]
```diff
@@ -90,8 +90,14 @@ func (v *fixedNameValidator) validate(ctx context.Context, obj runtime.Object, i
...
 	for _, dc := range subject.Spec.Datacenters {
-		if dc.Spec.Kubevirt != nil && (dc.Spec.Kubevirt.Images.EnableCustomImages) || (dc.Spec.Kubevirt.Images.HTTP != nil && dc.Spec.Kubevirt.Images.HTTP.ImageCloning.Enabled) {
-			return errors.New("this option is disabled for the Community Edition")
+		if dc.Spec.Kubevirt != nil {
+			if dc.Spec.Kubevirt.Images.EnableCustomImages {
+				return errors.New("custom images option is disabled for the Community Edition")
+			}
+
+			if dc.Spec.Kubevirt.Images.HTTP != nil && dc.Spec.Kubevirt.Images.HTTP.ImageCloning.Enabled {
+				return errors.New("image cloning options is disabled for the Community Edition")
```
Comment (hdurand0710) [inline pkg/webhook/seed/validation_ce.go:99]: s/options/option

File: pkg/resources/data.go [hunk 4]
```diff
@@ -785,6 +785,15 @@ func (data *TemplateData) GetEnvVars() ([]corev1.EnvVar, error) {
 	}
 	if cluster.Spec.Cloud.Kubevirt != nil {
 		vars = append(vars, corev1.EnvVar{Name: "KUBEVIRT_KUBECONFIG", ValueFrom: refTo(KubeVirtKubeconfig)})
+
+		if dc.Spec.Kubevirt != nil {
+			allowPVCClone := dc.Spec.Kubevirt.Images.HTTP != nil && dc.Spec.Kubevirt.Images.HTTP.ImageCloning.Enabled
+			vars = append(vars, corev1.EnvVar{Name: "KUBEVIRT_ALLOW_PVC_CLONE", Value: strconv.FormatBool(allowPVCClone)})
```
Comment (hdurand0710) [inline pkg/resources/data.go:791]: Will need to update with: `KUBEVIRT_ENABLE_PVC_CLONE` and `KUBEVIRT_PVC_CLONE_NAMESPACES` when machine-controller PR merged and bumped in this PR (or the next one ? )

**Comment by hdurand0710** [comment]
looks good !

**Pull request closed without merging**